src/modules/user/user.usecase.ts

```diff
@@ -385,6 +405,110 @@ export class UserUseCases {
     }
   }
 
+  async replacePreCreatedUser(
+    email: string,
+    newUserUuid: string,
+    newPublicKey: string,
+  ) {
+    const preCreatedUser = await this.preCreatedUserRepository.findByUsername(
+      email,
+    );
+
+    if (!preCreatedUser) {
+      return;
+    }
+
+    const defaultPass = this.configService.get('users.preCreatedPassword');
+    const { privateKey: encPrivateKey } = preCreatedUser;
+    const privateKey = aes.decrypt(encPrivateKey, defaultPass);
+
+    const invites = await this.sharingRepository.getInvitesBySharedwith(
+      preCreatedUser.uuid,
+    );
+
+    for (const invite of invites) {
+      const decryptedEncryptionKey = await decryptMessageWithPrivateKey({
+        encryptedMessage: Buffer.from(invite.encryptionKey, 'base64').toString(
```

Comment:
frontend side we are using aob and btoa but they are marked as deprecated by Nodejs, so used Buffer to accomplish the same behavior.  Same as below with buffer.from(xxx, 'binary')